tap_jira/tap.py

```diff
@@ -53,6 +55,25 @@ class TapJira(Tap):
                 ),
             ),
         ),
+        th.Property(
+            "stream_options",
+            th.ObjectType(
+                th.Property(
+                    "issues",
+                    th.ObjectType(
+                        th.Property(
+                            "jql",
+                            th.StringType,
+                            description="A JQL query to filter issues",
+                            title="JQL Query",
+                        ),
+                    ),
+                    title="Issues Stream Options",
+                    description="Options specific to the issues stream",
+                ),
+            ),
+            description="Options for individual streams",
+        ),
```

Comment:
@jeffgeorge I've decided to nest the setting in case we want to add similar options to other streams in the future, and it's a pattern that we recently started in tap-github.

Reply:
After this PR is merged and MeltanoHub is updated, you will be able to configure this in Meltano using, for example, this environment variable:

```dotenv
TAP_JIRA_STREAM_OPTIONS_ISSUES_JQL='project = FOO'
```

Reply:
> @jeffgeorge I've decided to nest the setting in case we want to add similar options to other streams in the future, and it's a pattern that we recently started in tap-github.

Ok cool, that works for me!